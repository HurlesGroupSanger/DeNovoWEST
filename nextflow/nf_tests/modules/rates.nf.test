nextflow_process {

    name "Test Process GFFUTILS_DB"
    script "modules/rates.nf"
    process "GFFUTILS_DB"

    test("Should create a gffutils database") {

        when {
            process {
                """
                input[0] = file("$baseDir/nf_tests/test_data/rates/create_gffutils_db/b37/input/Homo_sapiens.GRCh37.87.DDX3X_SCN5A.gff3")
                """
            }
        }

        then {
            //check if test case succeeded
            assert process.success
            //analyze trace file
            assert process.trace.tasks().size() == 1

            assert snapshot(process.out.gffutils_db_ch).match()

            // Check that the output example is still up to date
			with(process.out.gffutils_db_ch) {
				assert path(get(0)).md5 == path("$baseDir/../examples/rates/create_gffutils_db/b37/output/Homo_sapiens.GRCh37.87.DDX3X_SCN5A.gff3.db").md5
			}
        }
    }
}