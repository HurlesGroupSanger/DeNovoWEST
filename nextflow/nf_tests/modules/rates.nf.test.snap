{
    "Should create a gffutils database": {
        "content": [
            [
                "Homo_sapiens.GRCh37.87.DDX3X_SCN5A.gff3.db:md5,cbc348b9650e9d2093d59272f56d0bf2"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2025-01-13T19:05:21.443965759"
    }
}